src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -140,6 +141,14 @@ public Instance(
             this.refreshBeansPeriod = appConfig.getRefreshBeansPeriod();
         }
 
+        this.service = (String) instanceMap.get("service");
+        if (this.service == null && initConfig != null) {
```

Comment:
Since you check for empty string below do you want to do that here too?

Reply:
done!